src/layouts/Header/ThemeSwitcher/Container.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { THEMES } from '../../../constants/themes';
+import { setCurrentTheme } from '../../../redux/actions/themeActions';
+import ThemeSwitcher from './ThemeSwitcher';
+
+const ThemeSwitcherContainer = ({ theme, switchTheme }) => {
+  function changeTheme() {
+    const currentTheme = theme === THEMES.light ? THEMES.dark : THEMES.light;
+
+    switchTheme(currentTheme);
+  }
+
+  useEffect(() => {
+    localStorage.setItem('theme', theme);
```

Comment:
How it works?

Reply:
I moved setting data to localStorageService, removed useEffect and calling the function on changeTheme function.